src/components/dialogs/contingency-list/contingency-list-utils.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    CONTINGENCY_LIST_TYPE,
+    CONTINGENCY_NAME,
+    COUNTRIES_1,
+    COUNTRIES_2,
+    EQUIPMENT_IDS,
+    EQUIPMENT_TABLE,
+    EQUIPMENT_TYPE,
+    NAME,
+    NOMINAL_VOLTAGE_1,
+    NOMINAL_VOLTAGE_2,
+    SCRIPT,
+} from '../../utils/field-constants';
+import { ContingencyListType } from '../../../utils/elementType';
+import { prepareContingencyListForBackend } from '../contingency-list-helper';
+import {
+    saveExplicitNamingContingencyList,
+    saveFormContingencyList,
+    saveScriptContingencyList,
+} from '../../../utils/rest-api';
+import { getRangeInputEmptyDataForm } from '../../utils/rhf-inputs/range-input';
+import {
+    getCriteriaBasedSchema,
+    getCriteriaEmptyFormData,
+} from './criteria-based/criteria-based-utils';
+import yup from '../../utils/yup-config';
+import { getExplicitNamingSchema } from './explicit-naming/explicit-naming-form';
+
+export const DEFAULT_ROW_VALUE = {
+    [CONTINGENCY_NAME]: '',
+    [EQUIPMENT_IDS]: [],
+};
+export const DEFAULT_TABLE_ROWS = [
+    DEFAULT_ROW_VALUE,
+    DEFAULT_ROW_VALUE,
+    DEFAULT_ROW_VALUE,
+];
+
+export const CONTINGENCY_LIST_EQUIPMENTS = {
+    LINE: { id: 'LINE', label: 'Lines' },
```

Comment:
You can add a field called "fields" with the array of EquipmentDefinition in criteria-based-utils instead.
We avoid duplication and no need to update at 2 differents places when the equipments changes.

```suggestion
    LINE: { id: 'LINE', label: 'Lines', fields: [countries1, countries2, nominalVoltage1, nominalVoltage2] },
```

Reply:
done